src/rust/ide/lib/view/src/graph_editor/builtin/visualization/native/documentation.rs

```diff
@@ -0,0 +1,253 @@
+    }
+
+    fn init(self) -> Self {
+        self.reload_style();
+        self
+    }
+
+    /// Set size of the documentation view.
+    fn set_size(&self, size:Vector2) {
+        self.size.set(size);
+        self.reload_style();
+    }
+
+    /// Generate HTML documentation from documented Enso code.
+    fn gen_html_from(program:String) -> FallibleResult<String> {
+        let parser = parser::DocParser::new()?;
+        let output = parser.generate_html_docs(program);
+        Ok(output?)
+    }
+
+    /// Prepare data string for Doc Parser to work with after deserialization.
+    /// FIXME [MM]:  Removes characters that are not supported by Doc Parser yet.
+    ///              https://github.com/enso-org/enso/issues/1063
+    fn prepare_data_string(data_inner:&visualization::Json) -> String {
+        let data_str = serde_json::to_string_pretty(&**data_inner);
+        let data_str = data_str.unwrap_or_else(|e| format!("<Cannot render data: {}>", e));
+        let data_str = data_str.replace("\\n", "\n");
+        data_str.replace("\"", "")
+    }
+
+    /// Create a container for generated content and embed it with stylesheet.
+    fn push_to_dom(&self, content:String) {
+        let data_str = format!(r#"<div class="docVis">{}{}</div>"#, doc_style(), content);
+        self.dom.dom().set_inner_html(&data_str)
+    }
+
+    /// Receive data, process and present it in the documentation view.
+    fn receive_data(&self, data:&visualization::Data) -> Result<(),visualization::DataError> {
+        let data_inner = match data {
+            visualization::Data::Json {content} => content,
+            _                                   => todo!(),
+        };
+
+        let data_str   = ViewModel::prepare_data_string(data_inner);
+        let output     = ViewModel::gen_html_from(data_str);
+        let mut output = output.unwrap_or_else(|_| String::from(PLACEHOLDER_STR));
+        if output     == "" { output = String::from(PLACEHOLDER_STR) }
+        // FIXME [MM] : Because of how Doc Parser was implemented in Engine repo, there is need to
+        //              remove stylesheet link from generated code, that would otherwise point to
+        //              non-existing file, as now stylesheet is connected by include_str! macro, and
+        //              soon will be replaced by a style generator.
+        //              This hack will be removed when https://github.com/enso-org/enso/issues/1063
+        //              will land in Engine's repo, also fixing non-existent character bug.
+        let import_css = r#"<link rel="stylesheet" href="style.css" />"#;
+        let output     = output.replace(import_css, "");
+
+        self.push_to_dom(output);
+        Ok(())
+    }
+
+    /// Load an HTML file into the documentation view when user is waiting for data to be received.
+    /// TODO [MM] : This should be replaced with a EnsoGL spinner in the next PR.
+    fn load_no_doc_screen(&self) {
```

Comment:
the name of the function is misleading. What does "load no doc screen" mean? I understand it is not a "no doc", rather "waiting" or "awaiting" right?